app/src/main/cpp/src/components/transport_manager/src/bluetooth_le/ble_control_protocol.cc

```diff
@@ -0,0 +1,107 @@
+#include "transport_manager/bluetooth_le/ble_control_protocol.h"
+
+#include "utils/logger.h"
+#include "utils/jsoncpp_reader_wrapper.h"
+#include "utils/helpers.h"
+
+#include <map>
+#include <algorithm>
+
+namespace transport_manager {
+namespace transport_adapter {
+
+SDL_CREATE_LOG_VARIABLE("TransportManager")
+
+    namespace {
+        std::map<BleProtolActions, std::string> mapActionNames = {
+        {BleProtolActions::ON_DEVICE_CONNECTED, "ON_DEVICE_CONNECTED"},
+        {BleProtolActions::ON_DEVICE_DISCONNECTED, "ON_DEVICE_DISCONNECTED"}};
+
+        const char cDefaultValue[] = "_";
+        const char cActionKey[] = "action";
+        const char cNameKey[] = "name";
+        const char cAddressKey[] = "address";
+        const char cParamsKey[] = "params";
+    }
+
+    BleProtolActions BleControlProtocol::GetMessageActionType(const MsgData& message) {
+
+        utils::JsonReader reader_;
+        Json::Value value;
+        
+        if(reader_.parse((char*)(message.data()), &value)) {
```

Comment:
@VadymLuchko use static cast pls

Reply:
unfortunately static_cast helpless there due to  uint8_t and char are considered completely different types
we can use reinterpreted_cast<> here which is not better